### PR TITLE
Fix width of documentation popup

### PR DESCRIPTION
**Pull request by mattn**
When the popupmenu is wider filled with window width, documentation window become too narrow.

image

After this change:

image

### CODE REVIEW

**Comment by mattn** [comment]
FYI, I added borders for documentation popup.

**Comment by mattn** [comment]
Do you prefer to use `minwidth` and `minheight` ?

File: autoload/lsp/ui/vim/documentation.vim [hunk 1]
```diff
@@ -118,11 +118,14 @@ function! s:show_documentation(event) abort
     if l:right
         let l:line = a:event['row'] + 1
         let l:col = a:event['col'] + a:event['width'] + 1 + (a:event['scrollbar'] ? 1 : 0)
+        if l:col > &columns - 2
+            let l:col -= winwidth(0) / 2
```
Comment (prabirshrestha) [inline autoload/lsp/ui/vim/documentation.vim:122]: do we need float2nr here? we have seem multiple bugs on float values before.

**Review by prabirshrestha — APPROVED** [review]
LGTM. thanks for fixing this. this has been irritating me too.

we should also try to do the same for neovim if it isn't done.

can you explain what do you mean by `minwidth` and `minheight`? Do you also mean we need to add `maxwidth` and `maxheight`?

Ideally getting https://github.com/vim-jp/vital.vim/pull/748 would help a lot in simplifying our popup code.

File: autoload/lsp/ui/vim/documentation.vim (hunk 1 above)
Comment (mattn) [reply autoload/lsp/ui/vim/documentation.vim:122]: > can you explain what do you mean by minwidth and minheight? Do you also mean we need to add maxwidth and maxheight?

Yes, popup_create have attribute minwidth/minheight. This possibly can avoid either this issue.

Comment (prabirshrestha) [reply autoload/lsp/ui/vim/documentation.vim:122]: That sounds good.

Comment (mattn) [reply autoload/lsp/ui/vim/documentation.vim:122]: Hmm, minwidth and maxwidth may noy be good. It is not possible to calculate proper height and width. So I'll merge this workaround.

**Pull request merged**